tools/console/console.js

```diff
@@ -309,7 +309,7 @@ var ProgressDisplayFull = function (console) {
   self._progressBarRenderer = new ProgressBarRenderer(PROGRESS_BAR_FORMAT, options);
   self._progressBarRenderer.start = new Date();
 
-  self._headless = false;
+  self._headless = !!(process.env.METEOR_HEADLESS && process.env.METEOR_HEADLESS != '0');
```

Comment:
Is the `!= '0'` necessary?  If so, it should probably use `JSON.parse` in [the same manner](https://github.com/meteor/meteor/blob/devel/tools/cli/main.js#L803) as `METEOR_ALLOW_SUPERUSER`, but really, why not just have it work like most other env variables where if it's set in the environment, it's true otherwise, false: `!!(process.env.METEOR_HEADLESS)`

Either way, it should use strict equality `!==` not `!=`.